### PR TITLE
fix message in LHN when scanning

### PR DESCRIPTION
**Pull request by DylanDylann**
### Details

The subtitle in LHN display wrong message when requesting message by scanning

### Fixed Issues

$ https://github.com/Expensify/App/issues/27697
PROPOSAL: https://github.com/Expensify/App/issues/27697#issuecomment-1724054590

### Tests

1. Go to any chat
2. Clicking + button > Request Money > Scan
3. Upload receipt and request
4. See that in the LHN: Verify that the message is "Receipt scan in progress…"

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go to any chat
2. Clicking + button > Request Money > Scan
3. Upload receipt and request
4. See that in the LHN: Verify that the message is "Receipt scan in progress…"

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/141406735/3dc5e6fc-eac3-47c9-86df-5cf9a41e8eab

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/141406735/2c7e2c6a-6b92-4fbc-a497-16c1cae41d9a

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/141406735/e21a0309-096c-4b42-9e29-576e5cc45859

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/141406735/8a4753fc-3f48-46ef-af06-fbd43a8d7be6

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/141406735/072624c3-30e4-476d-9298-8f35425e43f7

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/141406735/c02044be-952e-4ff8-9179-2fe60e6d80bf

</details>

### CODE REVIEW

**Comment by cubuspl42** [comment]
@DylanDylann 

> Go to any chat

I don't think that a workspace chat is fine here, for example

**Comment by cubuspl42** [comment]
@DylanDylann You're probably right; let's forget about that.

I have a problem reproducing the fix in general:

https://github.com/Expensify/App/assets/2590174/4d08a943-d09f-4fa9-b114-58a47ac3b095

This is on your branch + merged `main`

**Comment by DylanDylann** [comment]
@cubuspl42 Thanks, I am going to investigate now. But It's seem another bug

**Comment by DylanDylann** [comment]
@cubuspl42 I just updated to fix [this case](https://github.com/Expensify/App/pull/28244#issuecomment-1736924395). But It is another bug with another RCA, solution even though it also happens when we request money by the amount (manually request). See this video

https://github.com/Expensify/App/assets/141406735/9970e0bc-4eb3-4b26-9b30-cb668c552a3e



The RCA is that we don't reset lastMessageTranslationKey when calling requestMoney API

**Comment by cubuspl42** [comment]
@DylanDylann So what you're saying is that we were unlucky and another bug got into `main` in the meantime, and for that reason your solution seemed to stop working after we merged `main`, right?

**Comment by DylanDylann** [comment]
Yes, It is my meaning. But I updated to fix that bug

File: src/libs/ReportUtils.js [hunk 1]
```diff
@@ -1508,15 +1508,17 @@ function getReportPreviewMessage(report, reportAction = {}, shouldConsiderReceip
     }
 
     if (shouldConsiderReceiptBeingScanned && ReportActionsUtils.isReportPreviewAction(reportAction)) {
-        const linkedTransaction = TransactionUtils.getTransaction(reportAction.childLastReceiptTransactionIDs);
+        const lastTransactionID = reportAction.childLastReceiptTransactionIDs.split(',')[0];
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/libs/ReportUtils.js:1517]: When there are many transactions in a report, childLastReceiptTransactionIDs will be "1231245654,123454222,1234545344,12346654"

because this logic
https://github.com/Expensify/App/blob/109e96c982352eea8862ddf08b14b29d2c45e56d/src/libs/ReportUtils.js#L2435

And in this case, we need to get the last transaction

**Comment by cubuspl42** [comment]
image

```
Uncaught TypeError: reportAction.childLastReceiptTransactionIDs is undefined
    getReportPreviewMessage ReportUtils.js:1488
    getLastMessageTextForReport OptionsListUtils.js:395
    getOptionData SidebarUtils.js:314
    optionItem OptionRowLHNData.js:96
    React 3
    OptionRowLHNData OptionRowLHNData.js:94
    React 16
    setWithOnyxState web.development.js:2641
    sendDataToConnection web.development.js:794
    connect web.development.js:902
    promise callback*connect web.development.js:889
    connectMappingToOnyx web.development.js:2714
    componentDidMount web.development.js:2560
    each Underscore
    componentDidMount web.development.js:2559
    React 12
    _updateCellsToRender index.js:435
    _taskHandle index.js:77
    handle index.js:25
    requestIdleCallback handler*runAfterInteractions/promise< index.js:23
    runAfterInteractions index.js:22
    timeoutHandle index.js:73
    setTimeout handler*schedule index.js:72
    _scheduleCellsToRenderUpdate index.js:1416
    componentDidUpdate index.js:1163
    React 12
    _updateCellsToRender index.js:435
    _taskHandle index.js:77
    handle index.js:25
    requestIdleCallback handler*runAfterInteractions/promise< index.js:23
    runAfterInteractions index.js:22
    timeoutHandle index.js:73
    setTimeout handler*schedule index.js:72
    _scheduleCellsToRenderUpdate index.js:1416
    componentDidUpdate index.js:1163
    React 6
ReportUtils.js:1488

```

**Comment by DylanDylann** [comment]
@cubuspl42 how you can reproduce this bug? I can't reproduce it

**Comment by DylanDylann** [comment]
Oh, I can reproduce it now

**Comment by DylanDylann** [comment]
The bug we got is because I am using childLastReceiptTransactionIDs to get the last transaction. It is incorrect because it only be last receipt transaction not the last transaction. So I updated the code to get last money report action of the IOU report and then  using it to get transaction

File: src/libs/actions/IOU.js [hunk 2]
```diff
@@ -154,6 +154,7 @@ function buildOnyxDataForMoneyRequest(
             value: {
                 ...chatReport,
                 lastReadTime: DateUtils.getDBTime(),
+                lastMessageTranslationKey: '',
```
Comment (cubuspl42) [inline src/libs/actions/IOU.js:157]: Why is this needed in this PR?

Comment (DylanDylann) [reply src/libs/actions/IOU.js:157]: I added it as mentioned [here](https://github.com/Expensify/App/pull/28244#issuecomment-1738672824)

Comment (cubuspl42) [reply src/libs/actions/IOU.js:157]: Oh, sorry. That's that key.

**Comment by cubuspl42** [comment]
Is this behavior expected? It looks buggy, as if something goes wrong...

https://github.com/Expensify/App/assets/2590174/99edce05-ab51-4198-9c12-782e5b7f06bf

Also, in console:

```
Warning: Failed prop type: Invalid prop `transaction.comment` of type `string` supplied to `MoneyRequestPreview`, expected `object`.
MoneyRequestPreview@webpack-internal:///./src/components/ReportActionItem/MoneyRequestPreview.js:168:101
withOnyx@webpack-internal:///./node_modules/react-native-onyx/dist/web.development.js:2500:9
withLocalize(Component)
MoneyRequestAction@webpack-internal:///./src/components/ReportActionItem/MoneyRequestAction.js:101:16
withNetwork(MoneyRequestAction)
withOnyx@webpack-internal:///./node_modules/react-native-onyx/dist/web.development.js:2500:9
div
View<@webpack-internal:///./node_modules/@expensify/react-native-web/dist/exports/View/index.js:52:19
div
View<@webpack-internal:///./node_modules/@expensify/react-native-web/dist/exports/View/index.js:52:19
div
View<@webpack-internal:///./node_modules/@expensify/react-native-web/dist/exports/View/index.js:52:19
ReportActionItemSingle@webpack-internal:///./src/pages/home/report/ReportActionItemSingle.js:121:24
```

Sounds worryingly related to what we do here

**Comment by DylanDylann** [comment]
@cubuspl42 This is expected. About console, I don't think it relates to this PR

**Comment by cubuspl42** [comment]
> Your video is expected

- I upload a receipt
- The chat summary goes from "cubuspl42.1+3... owes..." to "Scanning the receipt..." to "cubuspl42.1+3... owes..." and **again** to "Scanning the receipt..."
- Out of the blue, this appears, and it wasn't there before:
image

**Comment by DylanDylann** [comment]
@cubuspl42 I try to reproduce your bug but I can't. Could you try again?

https://github.com/Expensify/App/assets/141406735/b0dd6eac-ff1a-4b61-bbbc-c8bb022daab7

**Comment by cubuspl42** [comment]
> Out of the blue, this appears, and it wasn't there before

I believe that this is an unrelated thing, but this part...

> The chat summary goes from "cubuspl42.1+3... owes..." to "Scanning the receipt..." to "cubuspl42.1+3... owes..." and again to "Scanning the receipt..."

...might be related to our PR.

I'll test again.

**Comment by cubuspl42** [comment]
@DylanDylann I can't reproduce it with the last `main` merged-in. Maybe it was something unrelated, or a backend glitch.

**Comment by cubuspl42** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/2590174/8b3adfe8-8925-4776-8aac-edc7c503d07a

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/2590174/8a227474-4b69-462b-883a-945a5c9e1183



</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/2590174/6a14a529-fd21-42d4-9f01-cef4dc97fd73



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



https://github.com/Expensify/App/assets/2590174/c205d740-0774-47b1-b16c-b2d8f4dd2d18

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/2590174/6c9da9e5-bffe-4f1e-9f63-529bbfe436f0

</details>

**Review by jakub-trzebiatowski — APPROVED** [review]

**Comment by DylanDylann** [comment]
@tylerkaraszewski Gentle bump, please help to review this PR when you have a chance

**Comment by DylanDylann** [comment]
@tylerkaraszewski Friendly bump, Please check if you are available

**Review by tylerkaraszewski — APPROVED** [review]
